tutorials/07-voting/README.md

```diff
@@ -0,0 +1,893 @@
+        }
+
+        /// deposit takes a Vault object as an argument and adds
+        /// its balance to the balance of the owners Vault.
+        ///
+        /// It is allowed to destroy the sent Vault because the Vault
+        /// was a temporary holder of the tokens. The Vault's balance has
+        /// been consumed and therefore can be destroyed.
+        pub fun deposit(from: @FungibleToken.Vault) {
+            let vault <- from as! @VotingTutorialGovernanceToken.Vault
+            self.balance = self.balance + vault.balance
+            emit TokensDeposited(amount: vault.balance, to: self.owner?.address)
+            self.recordVotingWeight()
+            destroy vault
+        }
+
+        /// recordVotingWeight records the current voting weight of the vault
+        priv fun recordVotingWeight() {
+            let ts = getCurrentBlock().timestamp
+            self.votingWeightDataSnapshot.append(VotingWeightData(vaultBalance: self.balance, blockTs: ts))
+        }
+
+    }
+
+    /// createEmptyVault creates a new Vault with a balance of zero
+    /// and returns it to the calling context. A user must call this function
+    /// and store the returned Vault in their storage in order to allow their
+    /// account to be able to receive deposits of this token type.
+    pub fun createEmptyVault(): @FungibleToken.Vault {
+        return <-create Vault(balance: 0.0)
+    }
+
+    /// VaultMinter
+    ///
+    /// Resource object that an admin can control to mint new tokens
+    ///
+    pub resource VaultMinter {
+
+        /// mintTokens mints new tokens and deposits them into an account's vault
+        /// using their `Receiver` reference.
+        /// We say `&AnyResource{Receiver}` to say that the recipient can be any resource
+        /// as long as it implements the Receiver interface
+        pub fun mintTokens(amount: UFix64, recipient: Capability<&AnyResource{FungibleToken.Receiver}>) {
+            let recipientRef = recipient.borrow()
+                ?? panic("Could not borrow a receiver reference to the vault")
+
+            VotingTutorialGovernanceToken.totalSupply = VotingTutorialGovernanceToken.totalSupply + UFix64(amount)
+            recipientRef.deposit(from: <-create Vault(balance: amount))
+
+            emit TokensMinted(amount: amount)
+        }
+    }
+
+    /// The init function for the contract. All fields in the contract must
+    /// be initialized at deployment. This is just an example of what
+    /// an implementation could do in the init function. The numbers are arbitrary.
+    init() {
+        self.totalSupply = 0.0
+
+        // Assign paths
+        self.VaultStoragePath = /storage/CadenceVotingTutorialVotingTutorialGovernanceTokenVaultStoragePath
+        self.MinterStoragePath = /storage/CadenceVotingTutorialVotingTutorialGovernanceTokenMinterStoragePath
+        self.VaultPublicPath = /public/CadenceVotingTutorialVotingTutorialGovernanceTokenVaultPublicPath
```

Comment:
All of these have `VotingTutorial` twice in the path name. Make sure that gets fixed here and in the actual contract